### PR TITLE
kasan/lto: Turn off lto by default anyway

### PR DESCRIPTION
**Pull request by W-M-R**
*Note: Please adhere to [Contributing Guidelines](https://github.com/apache/nuttx/blob/master/CONTRIBUTING.md).*

## Summary

arm64/qemu: Enable lto compilation by default
kasan/lto: Turn off lto by default anyway

Of course, there is an error here, that is, the conditional judgment of
```
  ifeq ($(CONFIG_LTO_NONE),n)
    CFLAGS += -fno-lto
  endif
```
 is wrong, it should be judged as 
`"ifneq ($(CONFIG_LTO_NONE),)"`

## Impact

*Update this section, where applicable, on how change affects users,
 build process, hardware, documentation, security, compatibility, etc.*

## Testing

*Update this section with details on how did you verify the change,
 what Host was used for build (OS, CPU, compiler, ..), what Target was
 used for verification (arch, board:config, ..), etc. Providing build
 and runtime logs from before and after change is highly appreciated.*

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



This PR does **not** fully meet the NuttX requirements.  While it provides a summary of the changes, it lacks crucial information in the Impact and Testing sections.

Here's a breakdown:

* **Summary:**  Adequate, but could be improved by explicitly stating *why* enabling LTO by default for arm64/qemu is desirable and *why* it needs to be turned off by default for kasan/lto.  What problem does this solve? What are the benefits?  Clarifying the incorrect conditional and the proposed correction is good.

* **Impact:** Completely inadequate.  This section needs to be filled out.  Consider the following:
    * **Impact on user:** Will user behavior change? Will they need to reconfigure anything? Will performance be affected?
    * **Impact on build:**  Will build times change? Will disk space usage change?  Will any new dependencies be introduced?
    * **Impact on hardware:**  Does this change affect specific hardware configurations?
    * **Impact on documentation:** Does the documentation need to be updated to reflect this change?
    * **Impact on security:**  Are there any security implications of enabling/disabling LTO?
    * **Impact on compatibility:** Does this break compatibility with any existing configurations or tools?

* **Testing:** Completely inadequate. This section must be filled out with specific details:
    * **Build Host(s):**  Provide the OS, CPU architecture, and compiler version used for building.
    * **Target(s):** Specify the architecture, board, and configuration used for testing.
    * **Testing logs before change:** Include relevant logs demonstrating the behavior *before* the change. This could include build output, runtime output, or performance benchmarks.
    * **Testing logs after change:** Include corresponding logs demonstrating the behavior *after* the change, showing how the issue was resolved or the new feature functions.


**In short:** The PR needs significant expansion in the Impact and Testing sections before it can be properly reviewed.  Simply stating "Update this section..." is insufficient.  Provide concrete details and evidence to demonstrate the change's validity and impact.

**Comment by W-M-R** [comment]
This patch has nothing to do with the POSIX test of SIM, and will not cause ci error
image

File: mm/kasan/CMakeLists.txt [hunk 1]
```diff
@@ -22,12 +22,9 @@
 set(SRCS hook.c)
 
 if(CONFIG_MM_KASAN)
+  list(APPEND FLAGS -fno-lto)
```
Comment (anchao) [inline mm/kasan/CMakeLists.txt:25]: May break the GHS toolchain, Reference:
https://github.com/apache/nuttx/blob/master/arch/arm/src/cmake/ghs.cmake#L52
https://github.com/apache/nuttx/blob/master/libs/libc/assert/CMakeLists.txt#L26-L29

Comment (W-M-R) [reply mm/kasan/CMakeLists.txt:25]: But this citest configuration does not turn on kasan?

**Comment by xiaoxiang781216** [comment]
====================================================================================
Configuration/Tool: esp32-devkitc/nxdiag
2024-12-05 07:34:54
------------------------------------------------------------------------------------
  Cleaning...
  Configuring...
  Building NuttX...
fatal: could not read Username for 'https://github.com/': No such device or address
make[3]: *** [Makefile:83: espressif_prepare] Error 128
make[3]: Target 'context' not remade because of errors.
make[2]: *** [Makefile:55: /github/workspace/sources/apps/system/nxdiag_context] Error 2
make[2]: Target 'context_all' not remade because of errors.
make[1]: *** [Makefile:185: context] Error 2
make: *** [tools/Unix.mk:457: /github/workspace/sources/apps/.context] Error 2
make: Target 'all' not remade because of errors.
/github/workspace/sources/nuttx/tools/testbuild.sh: line 385: /github/workspace/sources/nuttx/../nuttx/nuttx.manifest: No such file or directory
  Normalize esp32-devkitc/nxdiag

**Pull request merged**